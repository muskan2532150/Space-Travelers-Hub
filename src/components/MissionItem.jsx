import React from 'react';
import { useDispatch } from 'react-redux';
import { missionChange } from '../redux/Mission/Mission';

export const MissionItem = ({ mission }) => {
  const dispatch = useDispatch();

  return (
    <section className="mission">
      <h3 className="mission">{mission.mission_name}</h3>
      <p className="mission-description border-helper">{mission.description}</p>
      <div className="mission-active-member-wrapper border-helper">
        <h3 className="mission-active-member ">
          {mission.mission ? 'Active Member' : 'Not A Member'}
        </h3>
      </div>
      <div className="border-helper mission-join-mission-wrapper">
        <button
          type="button"
          className="mission-join-mission"
          onClick={() => dispatch(missionChange(mission.mission_id))}>
          {mission.mission ? 'Leave Mission' : 'Join Mission'}
        </button>
      </div>
    </section>
  );
};
