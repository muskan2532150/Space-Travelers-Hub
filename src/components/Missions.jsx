import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { fetchdata } from "../redux/Mission/Mission";


const Missions = () => {
const dispatch =useDispatch();

useEffect(() => {
  dispatch(fetchdata());
});


  return (
    <div>
     
    </div>
  );
};

export default Missions;
