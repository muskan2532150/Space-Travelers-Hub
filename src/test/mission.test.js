import { render } from "@testing-library/react";
import Missions from "../components/MissionContainer";
import { Provider } from "react-redux";
import  store  from "../redux/Store";
import MissionItem from "../components/MissionItem";

const obj = {
    mission_name: 'muskan',
    mission_id: '1',
    description: 'desc',
    mission: false,
}

test('Demo test', () => {
  expect((1+2)).toBe(3);
});

test('renders MissionItem Component', ()=>{
  const missionItem = render(<Provider store={store}><MissionItem mission={obj} key={obj.mission_id} /></Provider>);
    expect(missionItem).toMatchSnapshot();
});

test('renders Mission Component', () => {
  const missions = render(<Provider store={store}><Missions /></Provider>);
  expect(missions).toMatchSnapshot();
});


