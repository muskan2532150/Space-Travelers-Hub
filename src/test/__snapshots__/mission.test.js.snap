// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Mission Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="all-missions"
      >
        <div
          class="missions-headings"
        >
          <h3
            class="mission-h3"
          >
            Mission
          </h3>
          <h3
            class="description-h3"
          >
            Description
          </h3>
          <h3
            class="status-h3"
          >
            Status
          </h3>
          <h3
            class="empty-h3"
          >
             
          </h3>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="all-missions"
    >
      <div
        class="missions-headings"
      >
        <h3
          class="mission-h3"
        >
          Mission
        </h3>
        <h3
          class="description-h3"
        >
          Description
        </h3>
        <h3
          class="status-h3"
        >
          Status
        </h3>
        <h3
          class="empty-h3"
        >
           
        </h3>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders MissionItem Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="mission"
      >
        <h3>
          muskan
        </h3>
        <p>
          desc
        </p>
        <div>
          <h3
            class="unactive"
          >
            Not A Member
          </h3>
        </div>
        <div>
          <button
            class=""
            type="button"
          >
            Join Mission
          </button>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="mission"
    >
      <h3>
        muskan
      </h3>
      <p>
        desc
      </p>
      <div>
        <h3
          class="unactive"
        >
          Not A Member
        </h3>
      </div>
      <div>
        <button
          class=""
          type="button"
        >
          Join Mission
        </button>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
